### PR TITLE
[sui-framework] Update doc generation to be a graph instead of a tree

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Post processes generated documentation for the framework so that we
* Don't have duplicate docs for shared dependencies between packages (e.g., `move-stdlib` for `sui-framework` and `sui-system`).
* Makes each package's documentation top-level
* Makes documentation a graph. 

Bottom commit are the actual changes, top commit are the regenerated docs. 

## Test Plan 

Regenerated, and then clicked around different links to make sure they worked as expected. 

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-framework/build.rs [hunk 1]
```diff
@@ -163,30 +164,53 @@ fn build_packages_with_move_config(
...
-            let mut dst_path = PathBuf::from(DOCS_DIR);
-            dst_path.push(framework_dir);
             dst_path.push(fname);
             fs::create_dir_all(dst_path.parent().unwrap()).unwrap();
             fs::write(dst_path, doc).unwrap();
         }
     }
 }
 
+fn relocate_docs(prefix: &str, files: &[(String, String)], output: &mut BTreeMap<String, String>) {
```
Comment (amnn) [inline crates/sui-framework/build.rs:192]: nit: docs explaining the what and why of this function would be helpful.

File: crates/sui-framework/build.rs [hunk 2]
```diff
@@ -163,30 +164,53 @@ fn build_packages_with_move_config(
...
+    let is_top_level = |path: &PathBuf| path.components().count() == 1;
+    for (file_name, file_content) in files {
+        let path = PathBuf::from(file_name);
+        let top_level = is_top_level(&path);
+        let file_name = if top_level {
```
Comment (amnn) [inline crates/sui-framework/build.rs:197]: nit (optional): inline `top_level`?

File: crates/sui-framework/build.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-framework/build.rs:192]: Also, this implementation is doing a depth-first traversal, IIUC, which means that currently the version of the `sui-framework` docs that get written will be the version that came from deepbook's dependencies. Is there value in ensuring that all the docs that get copied over are top-level docs, and then fixing up the links in those, or should it not matter?

**Review by amnn — APPROVED** [review]
Love it! Thanks @tzakian -- left some questions and nits, but this looks good to go to me.

File: crates/sui-framework/build.rs (hunk 1 above)
Comment (tzakian) [reply crates/sui-framework/build.rs:192]: It shouldn't matter. The only difference between them will be the relative links to things outside of the package which we're already changing.

**Review by ronny-mysten — APPROVED** [review]

**Pull request merged**